File: homework01/caesar.py
def encrypt_caesar(plaintext):
    """
    >>> encrypt_caesar("PYTHON")
    'SBWKRQ'
    >>> encrypt_caesar("python")
    'sbwkrq'
    >>> encrypt_caesar("Python3.6")
    'Sbwkrq3.6'
    >>> encrypt_caesar("")
    ''
    """
    ciphertext = ''
    for lit in plaintext:
        if 'A' <= lit <= 'Z' or 'a' <= lit <= 'z':
            if lit >= 'X' and lit <= 'Z' or lit >= 'x' and lit <= 'z':
                ciphertext += chr(ord(lit)+3-26)
            else:
                ciphertext += chr(ord(lit)+3)
        else:
            ciphertext += lit
    return ciphertext


def decrypt_caesar(ciphertext):
    """
    >>> decrypt_caesar("SBWKRQ")
    'PYTHON'
    >>> decrypt_caesar("sbwkrq")
    'python'
    >>> decrypt_caesar("Sbwkrq3.6")
    'Python3.6'
    >>> decrypt_caesar("")
    ''
    """
    plaintext = ''
    for lit in ciphertext:
        if 'A' <= lit <= 'Z' or 'a' <= lit <= 'z':
            if lit >= 'A' and lit <= 'C' or lit >= 'a' and lit <= 'c':
                plaintext += chr(ord(lit) - 3 + 26)
            else:
                plaintext += chr(ord(lit) - 3)
        else:
            plaintext += lit
    return plaintext

File: homework01/caesar.go
package caesar


func EncryptCaesar(plaintext string, shift int) string {
	var ciphertext string
	var symbol int
	ciphertext=""
	for i:=0; i<len(plaintext);i++{
		symbol=int(plaintext[i])
		if (int('A')<=symbol)&&(int('Z')>=symbol) || (int('a')<=symbol)&&(int('z')>=symbol){
			symbol+=shift%26
			
			if (symbol>int('Z')) && (symbol<int('a')) || (symbol>int('z')){
				symbol-=26
			}
		}	
		ciphertext+=string(symbol)
	}
	return ciphertext
}

func DecryptCaesar(ciphertext string, shift int) string {
	var plaintext string
	var symbol int
	plaintext=""
	for i:=0; i<len(ciphertext);i++{
		symbol=int(ciphertext[i])
		if (int('A')<=symbol)&&(int('Z')>=symbol) || (int('a')<=symbol)&&(int('z')>=symbol) {
			symbol -= shift % 26

			if (symbol < int('A')) || (symbol < int('a')) && (symbol > int('Z')) {
				symbol += 26
			}
		}
		plaintext+=string(symbol)
	}
	return plaintext
}
